### DISCUSSION TITLE
Opening a repository directly in GitHub/GitLab/etc.

### DISCUSSION THREAD

**Discussion by gitwittidbit** [coder/code-server#5074] [Q&A]
Hi,

is it possible, instead of cloning a repository, to open it directly on GitHub/GitLab et al.?

(I tried finding the GitHub Repositories extension in the code-server market place but couldn't find it.)

Thanks

**Comment by lg-nineoneninethreenine** [comment]
Could be. My idea is that if you install the repos extension onto code-server via VSIX, you can achieve this. But I would normally search around the command palette(<kbd>F1</kbd>).

**Comment by gitwittidbit** [comment]
The thing is: Gitea doesn't seem to have an extension of its own. And GitHub's extension doesn't seem to be available in the code-server marketplace. I was hoping that the GitHub extension might also work for Gitea (but I can't find it in the extension menu).

There is a GitLab extension. But this one seems to only support browisg of the remote repository without cloning. Not actually opening it.

**Reply by lg-nineoneninethreenine** [reply]
That is why I just told you to get the VSIX and shoot for install. Wait for some time, I'll post a video on it today or tomorrow. Busy schedule :/

**Comment by gitwittidbit** [comment]
ah sorry. I am new to VS Code and Code-Server. So I just didn't understand what you were saying.

So I now downloaded the GitHub Remote Repositories extension vsix file, uploaded it to my Code-Server server and tried "Install from VSIX". Unfortunately, code-server says that it cannot install this exension as it is not compatible :(

**Reply by lg-nineoneninethreenine** [reply]
No, use the CLI! The extension installer gui is stupid like me sometimes. Just run `code-server --install-extension extension-name.vsix` and if you are in the same directory as your extension, tab completion is easy(just hit the tab key).

**Comment by gitwittidbit** [comment]
Thank you for your patience!

I tried again - this time using the cli. But the result, sadly, is the same: 

```
Unable to install extension 'github.remotehub' as it is not compatible with VS Code '1.64.2'.
Failed Installing Extensions: file:///home/coder/GitHub.remotehub-0.31.2022040601.vsix
```

On the bright side: I just learned tax completion which I find hugely helpful!!! So thanks for that, too.

**Reply by lg-nineoneninethreenine** [reply]
Hmm that's weird. I think VS Code as well as code-server allow cloning powers out of the box. @jsjoeio I think this one's for you.

**Reply by jsjoeio** [reply]
Ah darn it! Compatibility issues are the worst. We're upgrading VS Code versions so this should be a problem in the next release 🤞🏼

**Reply by lg-nineoneninethreenine** [reply]
Well, darn it. Maybe we could best make out own GUI, that's best.
